### PR TITLE
Add coding style reference

### PR DESCRIPTION
**Pull request by adamkrawczyk**
Add coding standard. 

Related to PR #1 discussion.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+JavaScriptWrapImports: true
+KeepEmptyLinesAtTheStartOfBlocks: true
+MacroBlockBegin: ''
+MacroBlockEnd: ''
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: None
+ObjCBlockIndentWidth: 2
+ObjCBreakBeforeNestedBlockParam: true
+ObjCSpaceAfterProperty: true
+ObjCSpaceBeforeProtocolList: true
```
Comment (styczen) [inline .clang-format:82]:
```suggestion
```
Those are for Objective-C.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  AfterControlStatement: true
+  AfterEnum: true
+  AfterFunction: true
+  AfterNamespace: true
+  AfterStruct: true
+  AfterUnion: true
+  AfterExternBlock: true
+  BeforeCatch: true
+  BeforeElse: true
+  IndentBraces: false
```
Comment (styczen) [inline .clang-format:34]:
```suggestion
```
I would to not break braces into new line.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
```
Comment (styczen) [inline .clang-format:36]:
```suggestion
BreakBeforeBraces: Attach
```
This settings with previous removal of `BraceWrapping` settings would keep braces in the same line e.g.:
```cpp
namespace N {
enum E {
  E1,
  E2,
};

class C {
public:
  C();
};

bool baz(int i) {
  try {
    do {
      switch (i) {
      case 1: {
        foobar();
        break;
      }
      default: {
        break;
      }
      }
    } while (--i);
    return true;
  } catch (...) {
    handleError();
    return false;
  }
}

void foo(bool b) {
  if (b) {
    baz(2);
  } else {
    baz(5);
  }
}

void bar() { foo(true); }
} // namespace N
```

File: .clang-format [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+IndentPPDirectives: None
+IndentWidth: 2
+IndentWrappedFunctionNames: false
+JavaScriptQuotes: Leave
+JavaScriptWrapImports: true
+KeepEmptyLinesAtTheStartOfBlocks: true
+MacroBlockBegin: ''
+MacroBlockEnd: ''
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: None
```
Comment (styczen) [inline .clang-format:78]:
```suggestion
NamespaceIndentation: All
```
Would you reconsider indenting? It allows to see what is the context of the code.

**Review by styczen — COMMENTED** [review]
Left a few comments on `.clang-format`. Code of conduct looks good.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+- **Efficiency**: Ensure the code is efficient in terms of both time and space.
+- **Portability**: Write portable code that can run on multiple platforms without modification.
+
+## 2. Naming Conventions
+
+### 2.1 Variables
+
+- **Global Variables**: `g_global_variable`
+- **Member Variables**: `m_member_variable`
+- **Local Variables**: `m_some_local`
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:19]: Why do we want to name local variables like member variables?

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+### 3.3 Braces
+
+- Braces should be used for all control structures and placed on the same line as the control statement.
+
+```cpp
+if (condition) {
+  m_some_local = value;
+} else {
+  m_some_local = other_value;
+}
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:56]: Same for local variables naming.

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+### 3.4 Spacing
+
+Use a single space after keywords and around operators.
+
+```cpp
+for (int i = 0; i < n; ++i) {
+  m_some_local += i;
+}
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:66]: Same for local variables naming.

File: CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+const int kMaxIterations = 1000;
+```
+
+### 5.2 Use of auto
+
+- Prefer explicit types over auto, except when the type is obvious from the context or where it significantly improves code clarity.
+
+```cpp
+int m_some_local = 42;  // Preferred
+auto m_some_local = getAnswer();  // Acceptable if getAnswer's return type is clear
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:135]: Same for local variables naming.

File: CODE_OF_CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+- Prefer using smart pointers (e.g., `std::unique_ptr`, `std::shared_ptr`) over raw pointers.
+
+### 5.4 Error Handling
+
+- Always check for errors and handle them appropriately. Avoid using exceptions for control flow, avoid using return codes for error handling.
+
+```cpp
+try {
+  m_some_local = performOperation();
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:148]: Same for local variables naming.

File: CODE_OF_CONDUCT.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+}
+```
+
+### 5.5 Const Correctness
+
+- Use `const` wherever possible to indicate that a variable is not modified.
+
+```cpp
+int getAnswer() const {
+  return m_some_local;
```
Comment (styczen) [inline CODE_OF_CONDUCT.md:168]: Same for local variables naming.

**Review by styczen — COMMENTED** [review]
Questions about local variable naming.

File: CODE_OF_CONDUCT.md (hunk 5 above)
Comment (adamkrawczyk) [reply CODE_OF_CONDUCT.md:19]: this is clearly a typo

**Comment by adamkrawczyk** [comment]
Cahnged and applied suggestions i would like to merge this if there are no further points of concern

**Review by styczen — APPROVED** [review]

**Pull request merged**